internal/cluster/cluster.go

```diff
@@ -1399,32 +1403,106 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 }
 
-func (m *Manager) deleteClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) error {
-	log := logutil.FromContext(ctx, m.log)
+func (m *Manager) extractManifestFolderAndFilename(ctx context.Context, path string) (string, string, bool) {
+	parts := strings.Split(path, "/")
+	partsLength := len(parts)
+	if partsLength != 4 || parts[1] != "manifests" {
+		return "", "", false
+	}
+	folder := parts[2]
+	filename := parts[3]
+	return folder, filename, true
+}
+
+func (m *Manager) getClusterFilesForFolder(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) ([]string, error) {
 	path := filepath.Join(string(*c.ID), folder) + "/"
 	files, err := objectHandler.ListObjectsByPrefix(ctx, path)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to list files in %s", path)
 		m.log.WithError(err).Errorf(msg)
-		return common.NewApiError(
+		return nil, common.NewApiError(
 			http.StatusInternalServerError,
 			errors.Errorf(msg))
 	}
+	return files, err
+}
 
-	var failedToDelete []string
+func (m *Manager) ResetClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	log := logutil.FromContext(ctx, m.log)
+	files, err := m.getClusterFilesForFolder(ctx, c, objectHandler, "")
+	if err != nil {
+		return err
+	}
+	// ListClusterManifestsInternal only returns user generated manifests so we can use this to determine whether or not a manifest should be deleted on reset.
+	userManifests, err := m.manifestApi.ListClusterManifestsInternal(ctx, manifests.V2ListClusterManifestsParams{ClusterID: *c.Cluster.ID})
+	if err != nil {
+		return err
+	}
 	for _, file := range files {
-		//skip log and manifests deletion when deleting cluster files
-		if folder == "" && (strings.Contains(file, "logs") || strings.Contains(file, constants.ManifestFolder) || strings.Contains(file, constants.ManifestMetadataFolder)) {
-			continue
+		// Path is in the form <cluster-id>/manifests/<folder-name>/<file-name>
+		currentFolder, currentFile, isManifest := m.extractManifestFolderAndFilename(ctx, file)
+		if isManifest {
+			isSystemGeneratedManifest := true
+			for _, manifest := range userManifests {
+				if manifest.Folder == currentFolder && manifest.FileName == currentFile {
+					isSystemGeneratedManifest = false
+					break
+				}
+			}
+			if isSystemGeneratedManifest {
+				log.Debugf("Deleting cluster %s S3 file: %s", c.ID.String(), file)
+				_, err := objectHandler.DeleteObject(ctx, file)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (m *Manager) DeleteClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	return m.deleteClusterFilesInFolders(ctx, c, objectHandler, []string{"logs"})
+}
+
+func (m *Manager) deleteAllClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	folders := []string{"", "logs", constants.ManifestFolder, constants.ManifestMetadataFolder}
```

Comment:
Does `""` not contain all the other entries? If we know we're going to delete everything do we really need all the other specific folders?

Reply:
Old behaviour did this too, did not quite understand why, I am just replicating here but was curious about this too.